include/swift/Sema/ConstraintGraph.h

```diff
@@ -182,8 +197,13 @@ class ConstraintGraphNode {
   /// The type variable this node represents.
   TypeVariableType *TypeVar;
 
-  /// The set of bindings associated with this type variable.
-  inference::PotentialBindings Bindings;
+  /// The potential bindings for this type variable, updated incrementally by
+  /// the constraint graph.
+  inference::PotentialBindings Potential;
+
+  /// The binding set for this type variable, computed by
+  /// determineBestBindings().
+  std::optional<inference::BindingSet> Set;
```

Comment:
What is this actually buying us at the moment when it's reset during `determineBestBindings`?

Reply:
It avoids constructing the DenseMap `cache` in determineBestBindings().

Reply:
Is that really expensive? I'd expect that the majority of cost here is related to having to re-construct the binding sets...

Reply:
It is a significant speedup, yes. Agree that it's better to not have to reconstruct them at all, but one step at a time :)

Reply:
No worries, I'm just a bit surprised that building a map is that expensive...

Reply:
The change to the "sorting" part at the end where we track the pointer to the smallest binding set so far, instead of copying it, was a big win too :)